fhir-resource-tooling/src/main/java/com/ibm/cohort/tooling/fhir/MeasureImporter.java

```diff
@@ -0,0 +1,287 @@
+/*
+ * (C) Copyright IBM Corp. 2021
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package com.ibm.cohort.tooling.fhir;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.io.PrintStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+import org.hl7.fhir.instance.model.api.IBaseResource;
+import org.hl7.fhir.r4.model.Bundle;
+import org.hl7.fhir.r4.model.CanonicalType;
+import org.hl7.fhir.r4.model.Library;
+import org.hl7.fhir.r4.model.Measure;
+import org.hl7.fhir.r4.model.MetadataResource;
+import org.hl7.fhir.r4.model.RelatedArtifact;
+import org.hl7.fhir.r4.model.ResourceType;
+
+import com.beust.jcommander.JCommander;
+import com.beust.jcommander.Parameter;
+import com.beust.jcommander.internal.Console;
+import com.beust.jcommander.internal.DefaultConsole;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.ibm.cohort.fhir.client.config.FhirClientBuilderFactory;
+import com.ibm.cohort.fhir.client.config.FhirServerConfig;
+
+import ca.uhn.fhir.context.FhirContext;
+import ca.uhn.fhir.parser.IParser;
+import ca.uhn.fhir.rest.api.MethodOutcome;
+import ca.uhn.fhir.rest.client.api.IGenericClient;
+
+/**
+ * Load a FHIR server with resources provided in the UI measure export format. Resources are cross-referenced
+ * with existing content in the FHIR server by canonical url. Already existing resources are updated as needed.
+ * 
+ * Pre-reqs:
+ * $&lt; keytool -importkeystore -srckeystore /path/to/git/FHIR/fhir-server/liberty-config/resources/security/fhirKeyStore.p12 -srcstorepass change-password -cacerts -deststorepass changeit
+ */
+public class MeasureImporter {
+	private static final String DEPLOYPACKAGE = "deploypackage/";
+	private static final String JSON_EXT = ".json";
+
+	/**
+	 * General metadata about a deployable artifact.
+	 */
+	public abstract static class ArtifactMetadata {
+		public String resourceType;
+		public MetadataResource resource;
+		public String url;
+		public String version;
+		public String name;
+		public String id;
+
+		public void fromResource(MetadataResource metadata) {
```

Comment:
I find this class design to be somewhat confusing, in terms of contract.

From what I can tell, `LibraryArtifact:: updateDependencies` requires `resource` to be non-`null`,
but nothing is enforcing that you call `fromResource` before `updateDependencies`.

Maybe it makes sense to make this a Factory method instead?

Reply:
I updated the internal classes to have private constructors.